### PR TITLE
mplus-outline-fonts: add github release

### PR DESCRIPTION
**Pull request by este-rs**
###### Motivation for this change

[The new, revamped (pre)release](https://github.com/coz-m/MPLUS_FONTS) of the M+ font superfamily is highly deserving of inclusion, and is meant to replace the OSDN release for good. I've chosen to add a (sub)package instead of replacing the existing one so that we don't break existing user configurations (the fonts have been renamed and reshuffled across the two releases).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux, NixOS (with `--option sandbox true`)
- ~~Tested, as applicable~~ (inapplicable)
- ~~Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)~~ (nothing depends on this package)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @henrytill

### CODE REVIEW

**Review by henrytill — APPROVED** [review]
👍 Here

File: pkgs/data/fonts/mplus-outline-fonts/default.nix [hunk 1]
```diff
@@ -1,25 +1,67 @@
...
+
+  osdnRelease = makePackage rec {
+
+    fetcher = fetchzip;
+    variant = "OSDN";
+    license = lib.licenses.mit;
+
+    overrides = {
+
+      url = "mirror://osdn/mplus-fonts/62344/mplus-TESTFLIGHT-063a.tar.xz";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/mplus-outline-fonts/default.nix:28]:
```suggestion
in {
  osdnRelease = makePackage rec {
    fetcher = fetchzip;
    variant = "OSDN";
    license = lib.licenses.mit;

    overrides = {
      url = "mirror://osdn/mplus-fonts/62344/mplus-TESTFLIGHT-063a.tar.xz";
```

File: pkgs/data/fonts/mplus-outline-fonts/default.nix [hunk 2]
```diff
@@ -1,25 +1,67 @@
...
+        mkdir -p $out/share/fonts/truetype/${pname}
+        tar xvJf $downloadedFile
+        mv */*.ttf $out/share/fonts/truetype/${pname}
+      '';
+
+      sha256 = "16jirhkjs46ac8cdk2w4xkpv989gmz7i8gnrq9bck13rbil7wlzr";
+
+    };
+
   };
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/mplus-outline-fonts/default.nix:40]:
```suggestion
      sha256 = "16jirhkjs46ac8cdk2w4xkpv989gmz7i8gnrq9bck13rbil7wlzr";
    };
  };
```

File: pkgs/data/fonts/mplus-outline-fonts/default.nix [hunk 3]
```diff
@@ -1,25 +1,67 @@
...
+
+  githubRelease = makePackage {
+
+    fetcher = fetchFromGitHub;
+    variant = "GitHub";
+    license = lib.licenses.ofl;
+
+    overrides = {
+
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/mplus-outline-fonts/default.nix:49]:
```suggestion
  githubRelease = makePackage {
    fetcher = fetchFromGitHub;
    variant = "GitHub";
    license = lib.licenses.ofl;

    overrides = {
```

File: pkgs/data/fonts/mplus-outline-fonts/default.nix [hunk 4]
```diff
@@ -1,25 +1,67 @@
...
+        mv */fonts/otf/* $out/share/fonts/opentype/${pname}
+      '';
+
+      sha256 = "181dh03x1fi17lnckiv4dban08z4lphq0vgvj9ibf3adipb4bjb5";
+
+    };
+
+  };
+
 }
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/mplus-outline-fonts/default.nix:67]:
```suggestion
      '';

      sha256 = "181dh03x1fi17lnckiv4dban08z4lphq0vgvj9ibf3adipb4bjb5";
    };
  };
}
```

**Review by SuperSandro2000 — COMMENTED** [review]
Why do we need both releases?

**Comment by uakci** [comment]
Updated upstream and rebased on top of fresh nixpkgs; also added `recurseIntoAttrs` at the top-level.

@SuperSandro2000 Thank you for your lints. And to answer your question: the GitHub release of M+ actually contains a different set of font styles – under different names, too – which would make it a breaking change if we replaced the OSDN release with it. Many users depend on the old M+ font set to be available on their system and we wouldn’t want to force their hand in replacing the configuration files for their editors and whatnot. This way we at least get a “deprecation period” (though I don’t think I have the freedom to declare the OSDN release as deprecated here. A warning could be nice though).

**Review by henrytill — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -24084,7 +24084,8 @@ with pkgs;
 
   mph_2b_damase = callPackage ../data/fonts/mph-2b-damase { };
 
-  mplus-outline-fonts = callPackage ../data/fonts/mplus-outline-fonts { };
+  mplus-outline-fonts = recurseIntoAttrs
+    (callPackage ../data/fonts/mplus-outline-fonts { });
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:24088]:
```suggestion
  mplus-outline-fonts = recurseIntoAttrs (callPackage ../data/fonts/mplus-outline-fonts { });
```

Comment (uakci) [reply pkgs/top-level/all-packages.nix]: Applied, thanks once again.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 151705` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>mplus-outline-fonts.githubRelease</li>
<li>mplus-outline-fonts.osdnRelease</li>
</ul>
</details>

**Pull request merged**